src/Filter/Filter.php

```diff
@@ -26,32 +26,40 @@ abstract class Filter implements FilterInterface
     protected $name;
 
     /**
-     * @var mixed|null
+     * @var mixed
      */
     protected $value;
 
     /**
-     * @var array
+     * @var array<string, mixed>
      */
     protected $options = [];
 
     /**
-     * @var string
+     * @var string|null
      */
     protected $condition;
 
-    public function initialize($name, array $options = []): void
+    public function initialize(string $name, array $options = []): void
     {
         $this->name = $name;
         $this->setOptions($options);
     }
 
-    public function getName()
+    public function getName(): string
     {
+        if (null === $this->name) {
+            throw new \LogicException(sprintf(
+                'Seems like you didn\'t call `initialize()` on the filter `%s`. Did you create it through `%s::create()`?',
```

Comment:
If we must enforce this behavior, maybe we should declare `__construct()` method as private and final.

Reply:
Well the factory itself calls the construct. If I do that, it will not be able to

Reply:
Oh, I was thinking about `initialize()` as a singleton.

Reply:
This will need to be improved o. 3.x with a migration path and some deprecations first. Can you create an issue about it?

Reply:
I've created #6268 in order to track this. Please, feel free to edit the issue if you think there is something else we should cover there.

Reply:
Thank you